fix: incompatibility with skip_etkdg and tautomers

joanimato opened the pull request:
If multiple tautomers are generated, then skip_etkdg is disabled.

joanimato committed:
fix: incompatibility with skip_etkdg and tautomers

diogomart commented:
This is very likely the line that destroys existing positions:
https://github.com/forlilab/molscrub/blob/05f4d85be6f0d72ee876a413026c01991612f082/molscrub/protonate.py#L154-L156

Maybe we could find the MCS and copy the positions for the atoms in the MCS.

diogomart commented:
Example: https://github.com/forlilab/Meeko/blob/1ae73db81da15fd43731abec3ef27011aa7e0ac2/meeko/polymer.py#L193-L212

joanimato committed:
feat: mcs in skip_etkdg + constrained embedding.

diogomart committed:
fix: remove Hs with Chem.RemoveHs, not inside SDF parser

current RDKit version labels Hs as explicit when removing inside
parser, which makes some tautomer reactions fail, see
https://github.com/rdkit/rdkit/issues/9020

joanimato committed:
fix: create reference mol before removing Hs. This should not affect the tautomer rxns
